### DISCUSSION TITLE
What is `Undefined` ?

### DISCUSSION THREAD

**Discussion by rocky** [Mathics3/mathics-core#513] [Q&A] [answered]
Is it treated like value like `True` or `False` or `ComplexInfinity`? 

RIght now on input we treat it as a Global

```
mathics --full-form

Mathics 5.0.3dev0
.. 

In[1]:= True
System`True
Out[1]= True

In[2]:= Undefined
Global`Undefined
Out[2]= Undefined

In[3]:= ComplexInfinity
System`ComplexInfinity
Out[3]= ComplexInfinity
```

Is this the correct behavior? 

If this is not right and it should be in the `System` context should `Undefined` be declared as a `PredefinedSymbol` (that is a value that can't be reassigned like True, and ComplexInfinity.

**Comment by mmatera** [comment]
https://reference.wolfram.com/language/ref/Undefined.html

In WMA

```
n[1]:= ?? Undefined                                                            

Out[1]= Undefined is a symbol that represents a quantity with no defined value.


        Attributes[Undefined]={Protected}
```

**Reply by mmatera** [reply]
> https://reference.wolfram.com/language/ref/Undefined.html
> 
> In WMA
> 
> ```
> n[1]:= ?? Undefined                                                            
> 
> Out[1]= Undefined is a symbol that represents a quantity with no defined value.
> 
> 
>         Attributes[Undefined]={Protected}
> ```

So, it is different than ```System`True``` in the sense that you can unprotect it and assign a different value.

**Reply by rocky** [reply]
But is this supposed to be to have Global or System context?

**Reply by rocky** [reply]
Should it be treated as a constant like `ComplexInfinity`?

**Reply by mmatera** [reply, marked as answer]
The context should `System` and just has the attribute `Protected` so it is not a numeric constant.
